pkg/binding/bindingpolicy.go

```diff
@@ -88,6 +88,55 @@ func (c *Controller) syncBindingPolicy(ctx context.Context, bindingPolicyName st
 		logger.V(5).Info("Enqueued Binding for syncing, while handling BindingPolicy", "name", bindingPolicy.Name)
 		c.enqueueBinding(bindingPolicy.GetName())
 
+		statusCollectorNames, err := c.identifyMissingStatusCollectors(bindingPolicy)
+		if err != nil {
+			return fmt.Errorf("failed to identify missing StatusCollector(s) for BindingPolicy %s: %w", bindingPolicyName, err)
+		}
+
+		// If at least one StatusCollector object is missing, use a Condition to indicate the missing object and requeue the BindingPolicy,
+		// o.w. use the Condition to indicate all StatusCollector object(s) are available and proceed
+		if len(statusCollectorNames) > 0 {
+			logger.V(4).Info("Missing StatusCollector(s)", "statusCollectorNames", statusCollectorNames, "bindingPolicyName", bindingPolicyName)
+
+			scCondition := v1alpha1.BindingPolicyCondition{
+				Type:               v1alpha1.TypeStatusCollectorsAvailable,
+				Status:             corev1.ConditionFalse,
+				Reason:             v1alpha1.ReasonReconcileError,
+				LastTransitionTime: metav1.Now(),
+				Message:            fmt.Sprintf("Missing StatusCollector(s) %s", statusCollectorNames),
+			}
+
+			policyWithProposedStatus := bindingPolicy.DeepCopy()
+			policyWithProposedStatus.Status = v1alpha1.BindingPolicyStatus{
+				ObservedGeneration: bindingPolicy.Generation,
+				Conditions:         v1alpha1.SetCondition(bindingPolicy.Status.Conditions, scCondition),
+			}
+
+			_, err := c.bindingPolicyClient.UpdateStatus(ctx, policyWithProposedStatus, metav1.UpdateOptions{FieldManager: ControllerName})
+			if err != nil {
+				return fmt.Errorf("failed to update status for BindingPolicy %s: %w", bindingPolicyName, err)
+			}
+
+			return fmt.Errorf("failed to sync BindingPolicy %s because StatusCollector(s) %s are missing", bindingPolicyName, statusCollectorNames)
+		}
+
+		scCondition := v1alpha1.BindingPolicyCondition{
+			Type:               v1alpha1.TypeStatusCollectorsAvailable,
+			Status:             corev1.ConditionTrue,
+			Reason:             v1alpha1.ReasonReconcileSuccess,
+			LastTransitionTime: metav1.Now(),
```

Comment:
If this is not a change from the previous state of this Condition then LastTransitionTime should not be updated.

Reply:
Improved.